### PR TITLE
Pin HDF5 ...

### PR DESCRIPTION
**Pull request by ocefpaf**
... and fix make check.

### CODE REVIEW

File: recipe/build.sh [hunk 1]
```diff
@@ -1,26 +1,25 @@
 #!/bin/bash
 
+if [[ $(uname) == 'Darwin' ]]; then
```
Comment (pelson) [inline recipe/build.sh]: The other option:

````
LD_LIBRARY_PATH=foobar

if [[ $(uname) == 'Darwin' ]]; then
   DYLD_LIBRARY_PATH=$LD_LIBRARY_PATH
   unset LD_LIBRARY_PATH
fi

File: recipe/meta.yaml [hunk 2]
```diff
@@ -16,19 +16,19 @@ requirements:
...
+        - hdf5 1.8.15.1
         - libnetcdf
         - udunits2
-        - krb5
         - expat
+        - krb5
     run:
         - gsl
-        - hdf5
+        - hdf5 1.8.15.1
```
Comment (pelson) [inline recipe/meta.yaml]: Roll on https://github.com/conda/conda-build/pull/747

File: recipe/build.sh (hunk 1 above)
Comment (ocefpaf) [inline recipe/build.sh]: Definitely cleaner. But now I copy-and-pasted the current pattern everywhere...

Comment (ocefpaf) [inline recipe/build.sh]: Still there is the problem that some people recommend to set that variable only for the `make check`, so maybe the original pattern is not that bad.

**Pull request merged**